pallets/transfer-allowlist/src/lib.rs

```diff
@@ -0,0 +1,726 @@
+		/// To update a delay the delay has to be set to future modifiable.
+		/// then an update delay extrinsic called
+		pub fn add_allowance_delay(
+			origin: OriginFor<T>,
+			currency_id: T::CurrencyId,
+			delay: T::BlockNumber,
+		) -> DispatchResult {
+			let account_id = ensure_signed(origin)?;
+			let count_delay =
+				match Self::get_account_currency_restriction_count_delay(&account_id, &currency_id)
+				{
+					None => Ok(AllowanceMetadata {
+						allowance_count: 0,
+						current_delay: Some(delay),
+						modifiable_at: None,
+					}),
+					Some(
+						metadata @ AllowanceMetadata {
+							current_delay: None,
+							..
+						},
+					) => Ok(AllowanceMetadata {
+						current_delay: Some(delay),
+						..metadata
+					}),
+					Some(AllowanceMetadata {
+						current_delay: Some(_),
+						..
+					}) => Err(DispatchError::from(Error::<T>::DuplicateDelay)),
+				}?;
+
+			<AccountCurrencyTransferCountDelay<T>>::insert(&account_id, &currency_id, count_delay);
+			Self::deposit_event(Event::TransferAllowanceDelayAdd {
+				sender_account_id: account_id,
+				currency_id,
+				delay,
+			});
+			Ok(())
+		}
+
+		#[pallet::call_index(4)]
+		#[pallet::weight(10_000 + T::DbWeight::get().reads_writes(0, 1).ref_time())]
+		/// Updates an allowance delay, only callable is the delay has been set to allow future modifications and
+		/// the delay modifiable_at block has been passed.
+		pub fn update_allowance_delay(
+			origin: OriginFor<T>,
+			currency_id: T::CurrencyId,
+			delay: T::BlockNumber,
+		) -> DispatchResult {
+			let account_id = ensure_signed(origin)?;
+			let current_block = <frame_system::Pallet<T>>::block_number();
+			match Self::get_account_currency_restriction_count_delay(&account_id, &currency_id) {
+				None => Err(DispatchError::from(Error::<T>::NoMatchingDelay)),
+				Some(AllowanceMetadata {
+					current_delay: None,
+					..
+				}) => Err(DispatchError::from(Error::<T>::NoMatchingDelay)),
+				Some(AllowanceMetadata {
+					modifiable_at: None,
+					..
+				}) => Err(DispatchError::from(Error::<T>::DelayUnmodifiable)),
+				Some(AllowanceMetadata {
+					modifiable_at: Some(modifiable_at),
+					..
+				}) if modifiable_at > current_block => Err(DispatchError::from(Error::<T>::DelayUnmodifiable)),
+				Some(metadata) => {
+					<AccountCurrencyTransferCountDelay<T>>::insert(
+						&account_id,
+						&currency_id,
+						AllowanceMetadata {
+							current_delay: Some(delay),
+							// we want to ensure that after the delay is modified, it cannot be modified on a whim
+							// without another modifiable_at set.
+							modifiable_at: None,
+							..metadata
+						},
+					);
+					Self::deposit_event(Event::TransferAllowanceDelayUpdate {
+						sender_account_id: account_id,
+						currency_id,
+						delay,
+					});
+					Ok(())
+				}
+			}
+		}
+
+		#[pallet::call_index(5)]
+		#[pallet::weight(10_000 + T::DbWeight::get().reads_writes(0, 1).ref_time())]
```

Comment:
Is it intended to add benchmarks to these methods?

Reply:
Is this for the weight or the call index line?  The weight line is just an approximation rn, but we have an issue open for the restricted transfers for benchmarks.  The call_index is to make sure that the extrinsic call index never changes for the rpc calls (extra explanation just because I'm not sure which line was being referred to :) ).

Reply:
Yeah, I was referring to the `pallet::weight` line. Ok, thanks for explaining!